src/auth/context/auth-store.ts

```diff
@@ -18,11 +19,35 @@ export const useAuth = (): AuthContextType => {
 };
 
 export const useAuthLogic = () => {
-  const [username, setUsername] = useState("");
+  const [username, setUsername] = useState(() => {
+    const storedUsername = localStorage.getItem("username");
+    const expiration = localStorage.getItem("expiration");
+    if (storedUsername && expiration && new Date(expiration) > new Date()) {
+      return storedUsername;
+    }
+    return "";
+  });
 
   const login = (username: string) => {
```

Comment:
lets move the login logic to a new service file `login.service.ts` and export two methods inside this service (`login(), logout()`).

you can drop `useAuthLogic` hook, so we can replace it with service.

** We should put any cussinis logic into a new service file, to insure **(Single-Responsibility design principle)**